route/v1/types.go

```diff
@@ -145,8 +146,137 @@ type RouteSpec struct {
 	// +kubebuilder:validation:Enum=None;Subdomain;""
 	// +kubebuilder:default=None
 	WildcardPolicy WildcardPolicyType `json:"wildcardPolicy,omitempty" protobuf:"bytes,7,opt,name=wildcardPolicy"`
+
+	// httpHeaders defines policy for HTTP headers.
+	//
+	// If this field is empty, the default values are used.
+	//
+	// +optional
+	HTTPHeaders RouteHTTPHeaders `json:"httpHeaders,omitempty" protobuf:"bytes,9,opt,name=httpHeaders"`
+}
+
+// RouteHTTPHeaders defines policy for HTTP headers.
+type RouteHTTPHeaders struct {
+	// actions specifies actions to take on headers.
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set or deleted for TLS passthrough
+	// connections.
+	// Setting HSTS header is supported via another mechanism in Ingress.Spec.RequiredHSTSPolicies.
+	// Setting httpCaptureHeaders for the headers set via this API will not work.
+	// If spec.clientTLS, spec.httpHeaders.forwardedHeaderPolicy, spec.httpHeaders.uniqueId, spec.httpHeaders.headerNameCaseAdjustments is set
+	// and if you set the same header via this API then that header will get replaced with the value specified via this API.
+	// If cache-control is set using this API then the existing value of cache-control will get replaced.
```

Comment:
I don't think these settings apply to Route, e.g. spec.clientTLS, etc.

Reply:
The reason the Route API godoc mentions the `spec.clientTLS`, `spec.httpHeaders.forwardedHeaderPolicy`, `spec.httpHeaders.uniqueId`, and `spec.httpHeaders.headerNameCaseAdjustments` fields is that the someone using the Route API will want to know what to expect when the configuration on the IngressController says to do one thing and the configuration on the Route says to do something different.  Namely, the godoc should explain that custom headers specified on the route can overwrite the headers related to client TLS, `x-forwarded-*` headers, and so on.  

Maybe this would be clearer:
```go
	// The headers set via this API will not appear in access logs.  In
	// addition, any headers set via this API on the route will overwrite
	// headers set via spec.httpHeaders.actions on the IngressController, as
	// well as any headers related to the spec.clientTLS,
	// spec.httpHeaders.forwardedHeaderPolicy, spec.httpHeaders.uniqueId,
	// and spec.httpHeaders.headerNameCaseAdjustments IngressController
	// options.
```

Reply:
@candita, does my suggestion make sense?  

@miheer, https://github.com/openshift/api/compare/6dfd8ce82d5dbad4f7869f4534d54d6b36103ccb..d96a073a627e27efd6bca05050afc09c090db207#diff-4dfca44f3f68bff8b6381ff3fac676c6be96da9ec1d15224bbd70f839744eb40L167-R167 dropped this text from `RouteHTTPHeaders`.  It seems you deleted the godoc without commenting about the change, and maybe you didn't see my suggestion with the revised godoc.  Do you understand why I asked you to add the original godoc, and does my suggested revised godoc make sense to you?

Reply:
Okay, @Miciah's update makes sense.

Reply:
Yes this text make more sense

Reply:
I have made some modification in this as there was a repetition of the over-ride sentence.

Reply:
Did you push the changes?  I don't see them here: https://github.com/openshift/api/blob/8e4d36aaebb33d8bc4e10bd2e5cfc2efeb803ff3/route/v1/types.go#L160-L172

Reply:
yes.

Reply:
You pushed after I commented.  * grin *.  https://github.com/openshift/api/compare/8e4d36aaebb33d8bc4e10bd2e5cfc2efeb803ff3..3ed233511a8dd9484a47a184709f6a75d32ce70e added the changes: https://github.com/openshift/api/blob/3ed233511a8dd9484a47a184709f6a75d32ce70e/route/v1/types.go#L170-L174